snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/net4j/SnomedExportServerIndication.java

```diff
@@ -672,7 +701,94 @@ private SnomedMapSetSetting getSettingForRefSet(final String refSetId) {
 	}
 	
 	private void logActivity(final String message) {
-		LOGGER.info(message);
 		LogUtils.logExportActivity(LOGGER, userId, branchPath, message);
 	}
+	
+	private Collection<CodeSystemVersionEntry> getCodeSystemVersions() {
+		
+		if (extensionOnly) {
+			
+			return CodeSystemRequests.prepareSearchCodeSystemVersion()
+				.all()
+				.filterByCodeSystemShortName(codeSystemShortName)
+				.build(SnomedDatastoreActivator.REPOSITORY_UUID)
+				.execute(getEventBus())
+				.getSync()
+				.getItems();
+			
+		}
+		
+		return collectAllCodeSystemVersions(getCodeSystem(codeSystemShortName));
+	}
+	
+	private Collection<CodeSystemVersionEntry> collectAllCodeSystemVersions(CodeSystemEntry codeSystem) {
+		
+		Set<CodeSystemVersionEntry> results = newHashSet();
+		
+		List<CodeSystemVersionEntry> codeSystemVersions = CodeSystemRequests.prepareSearchCodeSystemVersion()
+				.all()
+				.filterByCodeSystemShortName(codeSystem.getShortName())
+				.build(SnomedDatastoreActivator.REPOSITORY_UUID)
+				.execute(getEventBus())
+				.getSync()
+				.getItems();
+		
+		results.addAll(codeSystemVersions);
+		
+		if (!codeSystem.getShortName().equals(SnomedTerminologyComponentConstants.SNOMED_SHORT_NAME)) {
+			
+			if (!Strings.isNullOrEmpty(codeSystem.getExtensionOf())) {
+				
+				Branch codeSystemBranch = RepositoryRequests.branching()
+					.prepareGet(codeSystem.getBranchPath())
+					.build(SnomedDatastoreActivator.REPOSITORY_UUID)
+					.execute(getEventBus())
+					.getSync();
+				 
+				final CodeSystemEntry parentCodeSystem = getCodeSystem(codeSystem.getExtensionOf());
```

Comment:
Note for the future: need to make sure that extension code systems have the `extensionOf` property correctly set.